common/djangoapps/third_party_auth/saml.py

```diff
@@ -0,0 +1,21 @@
+"""
+Slightly customized python-social-auth backend for SAML 2.0 support
+"""
+
+from social.backends.saml import SAMLIdentityProvider, SAMLAuth
+
+
+class SAMLAuthBackend(SAMLAuth):
+    """
+    Customized version of SAMLAuth that gets the list of IdPs from third_party_auth's list of
+    enabled providers.
+    """
+    name = "tpa-saml"
+
+    def get_idp(self, idp_name):
+        """ Given the name of an IdP, get a SAMLIdentityProvider instance """
+        from .provider import Registry  # Import here to avoid circular import
```

Comment:
This seems to be a flaw in the module layout.  Maybe the `Registry` class should be moved to a separate module or something, but this is out of scope for this PR.